File: leandro/Js/Prova/src/Questao.ts
export class Questao {
  private _enunciado: string;
  private _alternativas: string[];
  private _resposta: string;

  constructor(enunciado: string, alternativas: string[], resposta: string) {
    this._enunciado = enunciado;
    this._alternativas = alternativas;
    this._resposta = resposta;
  }

  public apresentar(): void {
    console.log();
    console.log(`Questão: ${this._enunciado}`);
    for (let i = 0; i < this._alternativas.length; i++) {
      const letra = String.fromCharCode(97 + i);
      console.log(`${letra}) ${this._alternativas[i]}`);
    }
  }

  public responder(respostaUsuario: "a" | "b" | "c" | "d"): any {
    if (respostaUsuario === this._resposta) {
      console.log("Resposta correta!");
    } else {
      console.log(`Resposta incorreta!\nLetra correta: ${this._resposta}`);
    }
  }
}

File: leandro/Prova/src/Questao.java
public class Questao {

  private String pergunta;
  private String[] opcoes;
  private String gabarito;

  public Questao(String pergunta, String[] opcoes, String gabarito) {
    this.pergunta = pergunta;
    this.opcoes = opcoes;
    this.gabarito = gabarito;
  }

  public void apresentar() {
    System.out.println(pergunta);
    for (int i = 0; i < opcoes.length; i++) {
      System.out.println((char) ('a' + i) + ") " + opcoes[i]);
    }
  }

  public void responder(String resposta) {
    if (resposta.equalsIgnoreCase(gabarito)) {
      System.out.println("Parabéns, você acertou!");
    } else {
      System.out.println("Ops, resposta incorreta...");
    }
  }
}
